app/models/piece.rb

```diff
@@ -11,6 +11,29 @@ class Piece < ActiveRecord::Base
 
   # Instance Methods
 
+  def move_to(end_coord)
+
+    #Check if move is vaild for Piece
+    return false unless self.valid_move?(end_coord)
+
+    #Capture logic
+    game.uncaptured_pieces.each do |piece|
+      if end_coord == piece.coordinate
+        if self.color != piece.color
+          piece.destroy
```

Comment:
Well when a pawn reaches the end it is prompted to what piece you want. So you can have two queens if you want. I figured we would create a new piece when a pawn gets promoted. But it's open for discussion